Added async to discover factories

molaco opened the pull request:
I have added a stream to process block logs asyncronously in discover_factories. I had to change some types and lifetimes and add some dependencies. I originally implemented this without using amms Factory and was a lot faster maybe i have messed sth up with the lifetimes.  I also did some implementations using fututres unordered but they did not seem to increase performance.

0xDmtri commented:
Looks good imho. Can I ask you why did u pick stream and not just tokio::spawn() and then collect and join handles with appropriate rate limit? Also probably will be better to use tracing::error!() than eprintln!.

molaco commented:
@0xDmtri i used the stream because i thought that maybe in the future the whole process from factory to pairs and pool info could be done in one go. But you probably need to filter the factories and pairs so i dont think its possible. I have changed it and used FututresUnordered instead because it seems a simpler and works great. I also added tracing for the task errors. Let me know if there are any issues.
